packages/eas-cli/src/build/android/version.ts

```diff
@@ -39,3 +118,39 @@ export async function maybeResolveVersionsAsync(
     };
   }
 }
+
+async function writeVersionsToBuildGradleAsync({
+  projectDir,
+  exp,
+}: {
+  projectDir: string;
+  exp: ExpoConfig;
+}): Promise<string> {
+  const buildGradle = readBuildGradle(projectDir);
+  if (!buildGradle) {
+    throw new Error('missing build gradle');
+  }
+  let updatedBuildGradle = AndroidConfig.Version.setVersionName(exp, buildGradle);
+  updatedBuildGradle = AndroidConfig.Version.setVersionCode(exp, updatedBuildGradle);
+  writeBuildGradle({ projectDir, buildGradle: updatedBuildGradle });
+  return updatedBuildGradle;
+}
+
+export function readBuildGradle(projectDir: string): string | undefined {
+  const buildGradlePath = AndroidConfig.Paths.getAppBuildGradleFilePath(projectDir);
+  if (!fs.pathExistsSync(buildGradlePath)) {
+    return undefined;
+  }
+  return fs.readFileSync(buildGradlePath, 'utf8');
```

Comment:
```suggestion
  return await fs.readFile(buildGradlePath, 'utf8');
```
Don't use sync functions in node if there are async counterparts and you really don't have to use sync functions.